app/controllers/rides_controller.rb

```diff
@@ -0,0 +1,66 @@
+# app/controllers/rides_controller.rb
+class RidesController < ApplicationController
+  # before_action :authenticate_user!
+
+  # GET /rides
+  def index
+    @rides = Ride.all
+    render json: @rides, each_serializer: RideSerializer, status: :ok
+  end
+
+  # GET /rides/:id
+  def show
+    @ride = Ride.find(params[:id])
+    render json: @ride
+  end
+
+  # POST /rides
+  def create
+    @ride = Ride.new(ride_params)
+    if @ride.save
+      render json: @ride, status: :created
+    else
+      render json: @ride.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /rides/:id
+  def update
+    @ride = Ride.find(params[:id])
+    if @ride.update(ride_params)
```

Comment:
Use update! it will raise exception if there are any error while updating record
```suggestion
    if @ride.update!(ride_params)
```